cmake: xtensa: use SoC name in path to toolchain

dcpleung opened the pull request:
Xtensa requires building a new toolchain for a specific SoC. By default xtools built Xtensa toolchains all have prefix of xtensa-zephyr-elf. In order to distinguish different toolchains, they are now placed in their own directories under their SoC name. This allows us to have multiple Xtensa toolchains targeting multiple SoCs.

This is in preparation for introducing additional Xtensa toolchains. Requires https://github.com/zephyrproject-rtos/sdk-ng/pull/111.

Requires SDK v0.11.

SebastianBoe commented on cmake/toolchain/zephyr/0.11/target.cmake:
```diff
@@ -1,36 +1,32 @@
 elseif("${ARCH}" STREQUAL "x86_64")
   list(APPEND TOOLCHAIN_C_FLAGS -mx32)
   list(APPEND TOOLCHAIN_LD_FLAGS -mx32)
 endif()
```
Please treat these variables as immutable, e.g.

```
if(ARCH STREQUAL xtensa)
set(CROSS_COMPILE bla bla)
else()
set(CROSS_COMPILE yada yada)
endif()

```

It is easier to debug and read when the variables are not variable.

dcpleung replied:
I have updated the file. Is this what you want?

dcpleung committed:
cmake: xtensa: use SoC name in path to toolchain

Xtensa requires building a new toolchain for a specific SoC.
By default xtools built Xtensa toolchains all have prefix of
xtensa-zephyr-elf. In order to distinguish different toolchains,
they are now placed in their own directories under their SoC
name. This allows us to have multiple Xtensa toolchains
targeting multiple SoCs.

The additional level in path name is introduced in SDK v0.11.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
sdk: add support for Zephyr SDK 0.11

This adds support for Zephyr SDK 0.11.* by simply copying
from 0.10.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
cmake: toolchain/xtools: update to align with latest sdk-ng

The target.cmake for xtools needs to be updated for changes in
sdk-ng repo. This is simply a copy from the target.cmake for
0.10 SDK.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
cmake: need to specify m32/m64 for x86_64 toolchain

Since x86_64-zephyr-elf is a multi-lib toolchain, m32/m64
need to be specified for the compiler to return the correct
library path when queried (e.g. --print-libgcc-file-name).
This affects the compile check done by CMake. Without these
flags, the compiler returns incorrect toolchain path (e.g.
requiring 64-bit libraries but returning 32-bit library path).
This also affects compiler flag checks for "-lstdc++". Incorrect
library path results in error when checking for "-lstdc++", and
this flag will not be used for the build. This results in
undefined references when compiling C++ code.

This creates target_x86.cmake to add the necessary flags for
CMake to use. The target_x86_64.cmake is also created to
mirror the same change.

Also removing the -m32 flags for host-gcc since we are not
building x86 targets with the host-gcc compiler.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung commented:
@jhedberg

dcpleung commented:
Rolled into https://github.com/zephyrproject-rtos/zephyr/pull/20221
